tests/Unit/Domain/Test_ElementSearchTree.cpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <optional>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/ElementLogicalCoordinates.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/ElementSearchTree.hpp"
+
+SPECTRE_TEST_CASE("Unit.Domain.ElementSearchTree", "[Domain][Unit]") {
+  const auto search_tree =
+      domain::index_element_ids<2>({{0, {{{1, 0}, {1, 0}}}},
+                                    {0, {{{1, 0}, {1, 1}}}},
+                                    {0, {{{1, 1}, {1, 0}}}},
+                                    {0, {{{2, 2}, {1, 1}}}},
+                                    {0, {{{2, 3}, {1, 1}}}},
+                                    {1, {{{1, 0}, {1, 0}}}}});
+  CHECK(
+      element_logical_coordinates<2>(
+          {domain::BlockId{0},
+           tnsr::I<double, 2, Frame::BlockLogical>{{{0.25, 0.5}}}},
+          search_tree) ==
+      std::make_pair(ElementId<2>{0, {{{2, 2}, {1, 1}}}},
+                     tnsr::I<double, 2, Frame::ElementLogical>{{{0.0, 0.0}}}));
+  CHECK(
+      element_logical_coordinates<2>(
+          {domain::BlockId{0},
+           tnsr::I<double, 2, Frame::BlockLogical>{{{-0.5, 0.5}}}},
+          search_tree) ==
+      std::make_pair(ElementId<2>{0, {{{1, 0}, {1, 1}}}},
+                     tnsr::I<double, 2, Frame::ElementLogical>{{{0.0, 0.0}}}));
+  CHECK(
+      element_logical_coordinates<2>(
+          {domain::BlockId{1},
+           tnsr::I<double, 2, Frame::BlockLogical>{{{-0.5, -0.5}}}},
+          search_tree) ==
+      std::make_pair(ElementId<2>{1, {{{1, 0}, {1, 0}}}},
+                     tnsr::I<double, 2, Frame::ElementLogical>{{{0.0, 0.0}}}));
+  CHECK_FALSE(element_logical_coordinates<2>(
+                  {domain::BlockId{1},
+                   tnsr::I<double, 2, Frame::BlockLogical>{{{0.5, 0.5}}}},
+                  search_tree)
+                  .has_value());
+  CHECK_FALSE(element_logical_coordinates<2>(
+                  {domain::BlockId{2},
+                   tnsr::I<double, 2, Frame::BlockLogical>{{{-0.5, -0.5}}}},
+                  search_tree)
+                  .has_value());
```

Comment:
I have no idea how this is testing the result of `index_element_ids` or why it's trying to use `element_logical_coordinates` to do so. This looks like a test of `element_logical_coordinates` not `index_element_ids`